File: binewton.c
#include <stdio.h>
#include <math.h>
void newton(double (*f)(double), double (*df)(double), double(x0), int n) {
	int i, j;
	
	for(i=0; i< n; i++) {
		double dfx0 = df(x0);
		if(dfx0 == 0) {
			break;
		}
		
		double xi = x0 - f(x0) /dfx0;
		printf("x_%d = %.16f\n", i+1, xi);
		x0 = xi;
	}
}

int main() {
	double pi = 3.14159265358;
	
	double f(double x) {
		return (pi * pow(x, 2)) * (((3 * 4.27) - x)/3) - 199.21;
	}
	
	double df(double x) {
		return 1.04719 * (25.62* x - 3*pow(x, 2));
	}
	double x0 = 1.23  ;
	int n = 5;
	
	newton(f, df, x0, n);
}

File: binewton.py
import math

def newton(f, df, x0, n):
  for _ in range(n):
    x0 -= f(x0) / df(x0)
    yield x0

def f(x):
  return math.sqrt(2*9.81*x) * math.tanh( (math.sqrt(2*9.81*x)/2*8.74)*5.87 ) - 6.55 

def df(x):
  return (2.21472 * math.tanh(1.48746 * math.sqrt(x) - 6.55))/math.sqrt(x) + 3.29432 * pow(math.sech(1.48746 * math.sqrt(x) - 6.55),2)


x0 = 1.23
n = 10

resp = list(newton(f, df, x0, n))

iter = [1, 3, 5]



for i in iter:
  print(str(resp[i-1])+',', end="") 
